pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -482,6 +482,12 @@ func (gce *GCECloud) EnsureLoadBalancer(name, region string, requestedIP net.IP,
 		if err != nil {
 			return nil, err
 		}
+		defer func() {
+			if err := gce.deleteStaticIP(name, region); err != nil {
```

Comment:
If the delete fails, I don't think we'll actually come back into this function until the service is updated. Ideally any failure to create/delete/update a cloud resource should requeue the service in the controller, for a retry. It feels like we should return the error. 

Of course that would mean if the deletion fails because something else is using the IP, the controller will get wedged repeatedly trying to delete it.